zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java

```diff
@@ -205,6 +194,19 @@ public void export(final Record<?> record) {
     }
   }
 
+  private void ensureCachedRecordsLessThanBulkSize() {
+    final var maxCachedRecords = configuration.getBulk().getSize();
+
+    if (writer.getBatchSize() > maxCachedRecords) {
+      final var errMsg =
+          String.format(
+              "Reached the max bulk size amount of cached records [%d] while waiting for importers to finish, restarting exporter",
+              maxCachedRecords);
```

Comment:
Not sure what you mean by `error loop here for the last record`?

Reply:
I think Chris was hinting at how we handle an exception in `export(record)`. We don't restart the exporter in case of exception. but rather the `ExporterContainer` will return `false` for this specific record we failed at
https://github.com/camunda/camunda/blob/c3b85b4812a7001bf53504bb11789f1e667d9152/zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterContainer.java#L218-L232

And in the `ExporterDirector` we run this method with retry (with a backoff strategy)
https://github.com/camunda/camunda/blob/c3b85b4812a7001bf53504bb11789f1e667d9152/zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java#L654-L655

So in practice: we are (error) looping on exporting this last record that caused this exception